app/utils/auth.js

```diff
@@ -0,0 +1,70 @@
+module.exports = {
+    login(username, pass, cb) {
+        cb = arguments[arguments.length - 1]
+        if (sessionStorage.token) {
+            if (cb) cb(true)
+            this.onChange(true)
+            return
+        }
+
+        request(username, pass, (res) => {
+            if (res.authenticated) {
+                sessionStorage.token = res.token
+                if (cb) cb(true)
+                this.onChange(true)
+            } else {
+                if (cb) cb(false)
+                this.onChange(false)
+            }
+        })
+    },
+
+    getToken: function () {
+        return sessionStorage.token
+    },
+
+    logout: function (cb) {
+        delete sessionStorage.token
+        if (cb) cb()
+        this.onChange(false)
+    },
+
+    loggedIn: function () {
+        return !!sessionStorage.token
+    },
+
+    onChange: function () {}
+}
+
+function request(username, pass, cb) {
+
+    setTimeout(() => {
```

Comment:
Can we unwrap this code from `setTimeout()` here?

I think the `setTimeout` wrapper was an artifact from the auth-flow example that they used to simulate an async authentication operation.